clusters/clusters_api.go

```diff
@@ -298,6 +359,47 @@ type Cluster struct {
 	IdempotencyToken string `json:"idempotency_token,omitempty" tf:"force_new"`
 }
 
+func (cluster Cluster) Validate() error {
+	// TODO: rewrite with CustomizeDiff
+	if cluster.NumWorkers > 0 || cluster.Autoscale != nil {
+		return nil
+	}
+	profile := cluster.SparkConf["spark.databricks.cluster.profile"]
+	master := cluster.SparkConf["spark.master"]
+	resourceClass := cluster.CustomTags["ResourceClass"]
+	if profile == "singleNode" && strings.HasPrefix(master, "local") && resourceClass == "SingleNode" {
+		return nil
+	}
+	return fmt.Errorf("NumWorkers could be 0 only for SingleNode clusters. See https://docs.databricks.com/clusters/single-node.html for more details")
+}
+
+// ModifyRequest helps remove all request fields that should not be submitted when instance pool is selected.
+func (cluster *Cluster) ModifyRequest() {
```

Comment:
Maybe give it better name? like `modifyClusterOnInstancePool`, or something like